src/chart-addons/stacked-legend.js

```diff
@@ -119,83 +166,99 @@ export function getLegendStateFromChart(chart, useMap, selectedLayer) {
+          ] = layer.getPrimaryColorScaleAndLegend()
+          if (color_scale === null) {
+            return {}
+          }
+          const color_scale_name = color_scale.name || ""
+          const materialized_color_scale = vega.scales.find(
+            scale => scale.name === color_scale_name
+          )
+          if (!materialized_color_scale) {
+            return {}
+          }
+
+          return {
+            ...materialized_color_scale,
+            legend: color_legend,
+            legend_position,
+            version: 2.0
+          }
+        } else {
+          // TODO(croot): this can be removed once all raster layer types are
+          // transitioned to the getPrimaryColorScaleName/getLegendDefinitionForProperty
+          // form above
+          const layerState = layer.getState()
+          const color = layer.getState().encoding.color
+          // TODO[C]: the line above is where it is getting everything for the legend, including colors, domain and range
+
+          let color_legend_descriptor = null
 
-        return {
-          ...materialized_color_scale,
-          legend: color_legend,
-          legend_position,
-          version: 2.0
-        }
-      } else {
-        // TODO(croot): this can be removed once all raster layer types are
-        // transitioned to the getPrimaryColorScaleName/getLegendDefinitionForProperty
-        // form above
-        const layerState = layer.getState()
-        const color = layer.getState().encoding.color
-
-        let color_legend_descriptor = null
-
-        if (
-          (layers.length > 1 || _.isEqual(selectedLayer, layerState)) &&
-          typeof color !== "undefined"
-        ) {
           if (
-            typeof color.scale === "object" &&
-            color.scale.domain === "auto"
+            (layers.length > 1 || _.isEqual(selectedLayer, layerState)) &&
+            typeof color !== "undefined"
           ) {
-            color_legend_descriptor = {
-              ...color,
-              scale: {
-                ...color.scale,
+            if (
+              typeof color.scale === "object" &&
+              color.scale.domain === "auto"
+            ) {
+              color_legend_descriptor = {
+                ...color,
+                scale: {
+                  ...color.scale,
+                  domain: layer.colorDomain()
+                }
+              }
+            } else if (
+              typeof color.scale === "undefined" &&
+              color.domain === "auto"
+            ) {
+              color_legend_descriptor = {
+                ...color,
                 domain: layer.colorDomain()
               }
-            }
-          } else if (
-            typeof color.scale === "undefined" &&
-            color.domain === "auto"
-          ) {
-            color_legend_descriptor = {
-              ...color,
-              domain: layer.colorDomain()
+            } else if (color.type === "ordinal") {
```

Comment:
Ugly diff, due to the now-necessary `Promise.all` async wrapper.  Just did the async handling and added the condition below: